power_grid_model_c/power_grid_model/include/power_grid_model/main_model.hpp

```diff
@@ -921,6 +925,56 @@ class MainModelImpl<ExtraRetrievableTypes<ExtraRetrievableType...>, ComponentLis
         }
         return math_param;
     }
+    template <bool sym> std::vector<MathModelParamIncrement> get_math_param_increment() {
+        using AddToIncrement = void (*)(std::vector<MathModelParamIncrement>&, MainModelState const&, Idx2D const&);
+
+        static constexpr std::array<AddToIncrement, n_types> add_to_increments{
+            [](std::vector<MathModelParamIncrement>& increments, MainModelState const& state,
+               Idx2D const& changed_component_idx) {
+                if constexpr (std::derived_from<ComponentType, Branch>) {
+                    Idx2D const math_idx =
+                        state.topo_comp_coup->branch[state.components.template get_seq<Branch>(changed_component_idx)];
+                    if (math_idx.group == -1) {
+                        return;
+                    }
+                    // assign parameters
+                    increments[math_idx.group].branch_param_to_change.push_back(math_idx.pos);
+                } else if constexpr (std::derived_from<ComponentType, Branch3>) {
+                    Idx2DBranch3 const math_idx =
+                        state.topo_comp_coup
+                            ->branch3[state.components.template get_seq<Branch3>(changed_component_idx)];
+                    if (math_idx.group == -1) {
+                        return;
+                    }
+                    // assign parameters, branch3 param consists of three branch parameters
```

Comment:
This three lines are commented out because `branch3_param` is not used. Can you confirm this? @mgovers

Reply:
indeed. this is something i've also been looking to move to `main_core/*` and out of the main model itself like the `main_core/update.hpp` but i haven't got to this part yet. At that point, we should revisit but definitely out of scope of this PR. this is mainly just copied from the non-incremental one